js/try.js

```diff
@@ -214,11 +286,31 @@ const fetchAndParseTimetable = async (url) => {
       .replace(/"/g, '\'')
       .trim()
 
+  const response = await fetch(url)
+  if (!response.ok) {
+    throw new Error('Can\'t fetch timetable')
+  }
+  const text = await response.text()
+
+  const parser = new DOMParser()
+  const root = parser.parseFromString(text, 'text/html').body
+  const timetableElements = Array.from(
+    root.querySelectorAll('table.spreadsheet')
+  )
+
+  let date = false;
```

Comment:
Not sure what the point of this is? You can use the `baseDate` and the values of `days` to get the date of the class

Reply:
as mentioned above, `days` can be unreliable if the holidays change

also, this sort of math is a pain to work with... why would you do weird date arithmetic when you can use a simple Date object for each class?

Reply:
The original implementation was using Date objects, but this approach worked best for a calendar display

Reply:
This approach also doesn't have to handle date issues, as date and time are stored separately. With Date objects, the relevant information also has to be parsed and extracted, which makes the logic messier to work with.